cmd/server-main.go

```diff
@@ -158,6 +157,14 @@ func serverMain(ctx *cli.Context) {
 	globalPublicCerts, globalRootCAs, globalTLSCertificate, globalIsSSL, err = getSSLConfig()
 	fatalIf(err, "Invalid SSL certificate file")
 
+	// Check if certficates are available, and endpoints are secured as well. We
+	// should validate this only under distributed setup since distributed setup
+	// can have multiple secure endpoints. We shouldn't validate and fail for any
+	// other mode.
+	if !globalIsSSL && globalEndpoints.IsHTTPS() && globalIsDistXL {
```

Comment:
You could order conditions like `globalIsDistXL && globalEndpoints.IsHTTPS() && !globalIsSSL` to convey the flow.

Reply:
Done